src/Symfony/Component/Console/Attribute/Argument.php

```diff
@@ -0,0 +1,93 @@
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+
+#[\Attribute(\Attribute::TARGET_PARAMETER)]
+class Argument
+{
+    private ?int $mode = null;
+
+    /**
+     * Represents a console command <argument> definition.
+     *
+     * If unset, the `name` and `default` values will be inferred from the parameter definition.
+     *
+     * @param string|bool|int|float|array|null                               $default         The default value (for InputArgument::OPTIONAL mode only)
+     * @param array|callable-string(CompletionInput):list<string|Suggestion> $suggestedValues The values used for input completion
+     */
+    public function __construct(
+        public string $name = '',
+        public string $description = '',
+        public string|bool|int|float|array|null $default = null,
+        public array|string $suggestedValues = [],
+    ) {
+        if (\is_string($suggestedValues) && !\is_callable($suggestedValues)) {
+            throw new \TypeError(\sprintf('Argument 4 passed to "%s()" must be either an array or a callable-string.', __METHOD__));
+        }
+    }
+
+    public static function tryFrom(\ReflectionParameter $parameter): ?self
+    {
+        /** @var self $self */
+        if (null === $self = ($parameter->getAttributes(self::class, \ReflectionAttribute::IS_INSTANCEOF)[0] ?? null)?->newInstance()) {
+            return null;
+        }
+
+        $type = $parameter->getType();
+        $name = $parameter->getName();
+
+        if (!$type instanceof \ReflectionNamedType) {
+            throw new LogicException(\sprintf('The parameter "$%s" must have a named type. Untyped, Union or Intersection types are not supported for command arguments.', $name));
+        }
+
+        $parameterTypeName = $type->getName();
+
+        if (!\in_array($parameterTypeName, ['string', 'bool', 'int', 'float', 'array'], true)) {
+            throw new LogicException(\sprintf('The type "%s" of parameter "$%s" is not supported as a command argument. Only "string", "bool", "int", "float", and "array" are allowed.', $parameterTypeName, $name));
+        }
+
+        if (!$self->name) {
+            $self->name = $name;
+        }
+
+        $self->mode = null !== $self->default || $parameter->isDefaultValueAvailable() ? InputArgument::OPTIONAL : InputArgument::REQUIRED;
+        if ('array' === $parameterTypeName) {
+            $self->mode |= InputArgument::IS_ARRAY;
+        }
+
+        $self->default ??= $parameter->isDefaultValueAvailable() ? $parameter->getDefaultValue() : null;
+
+        if (\is_array($self->suggestedValues) && !\is_callable($self->suggestedValues) && 2 === \count($self->suggestedValues) && ($instance = $parameter->getDeclaringFunction()->getClosureThis()) && $instance::class === $self->suggestedValues[0] && \is_callable($instance::class, $self->suggestedValues[1])) {
+            $self->suggestedValues = [$instance, $self->suggestedValues[1]];
+        }
```

Comment:
[note for reviewer] this falls back from the "static class method call" syntax to the "object method call" syntax due to the impossibility of passing a `\Closure` or `callable` in the attribute constructor. Allowing this suggestion methods to access the instance's dependencies.

Reply:
For instance, this will allow us to configure `#[Argument(suggestedValues: [self::class, 'getPermissions'])]` where `getPermissions` is not defined as static method, enabling it to access any instance dependencies and dynamically retrieve all available permissions from another service (e.g. entity manager)